From fb587c87c52d139f3011ac13f388b9df3ccc9fea Mon Sep 17 00:00:00 2001
From: Oleksii Kurochko <oleksii.kurochko@gmail.com>
Date: Tue, 4 Nov 2025 09:40:33 +0100
Subject: [PATCH] xenstored: add extra logs around domain introduction

By these logs I want to understand why
xengnttab_map_grant_ref(...,GNTTAB_RESERVED_XENSTORE,...) isn't called.

Signed-off-by: Oleksii Kurochko <oleksii.kurochko@gmail.com>
---
 tools/xenstored/domain.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/tools/xenstored/domain.c b/tools/xenstored/domain.c
index 64c8fd0cc3..02e8c134f6 100644
--- a/tools/xenstored/domain.c
+++ b/tools/xenstored/domain.c
@@ -501,6 +501,9 @@ static const struct interface_funcs domain_funcs = {
 
 static void *map_interface(domid_t domid)
 {
+	syslog(LOG_ERR, "%s: domid(%d) master_domid(%d)\n",
+		   __func__, domid, xenbus_master_domid());
+
 	if (domid == xenbus_master_domid())
 		return xenbus_map();
 
@@ -970,10 +973,15 @@ static struct domain *introduce_domain(const void *ctx,
 	struct xenstore_domain_interface *interface;
 	bool is_master_domain = (domid == xenbus_master_domid());
 
+	syslog(LOG_ERR, "%s: domid(%d), is_master_domain(%d)\n", __func__, domid, is_master_domain);
+
 	domain = find_or_alloc_domain(ctx, domid);
 	if (!domain)
 		return NULL;
 
+	syslog(LOG_ERR, "%s: domid(%d)->introduced: %d\n",
+	       __func__, domid, domain->introduced);
+
 	if (!domain->introduced) {
 		interface = map_interface(domid);
 		if (!interface && !restore)
@@ -1003,6 +1011,8 @@ static struct domain *introduce_domain(const void *ctx,
 		domain->port = (rc == -1) ? 0 : rc;
 	}
 
+	syslog(LOG_ERR, "%s: domid(%d): %p\n", __func__, domid, domain);
+
 	return domain;
 }
 
@@ -1015,6 +1025,8 @@ int do_introduce(const void *ctx, struct connection *conn,
 	unsigned int domid;
 	evtchn_port_t port;
 
+	syslog(LOG_ERR, "%s is called\n", __func__);
+
 	if (get_strings(in, vec, ARRAY_SIZE(vec)) < ARRAY_SIZE(vec))
 		return EINVAL;
 
-- 
2.51.1

